frontend/src/components/viewers/MetricsVisualizations.tsx

```diff
@@ -36,44 +37,56 @@ interface MetricsVisualizationsProps {
  * and multiple visualizations associated with one artifact.
  */
 export function MetricsVisualizations({ artifacts, artifactTypes }: MetricsVisualizationsProps) {
-  // system.ClassificationMetrics contains confusionMatrix or confidenceMetrics.
-  // TODO: Visualize confusionMatrix using system.ClassificationMetrics artifacts.
-  // https://github.com/kubeflow/pipelines/issues/5668
-  let classificationMetricsArtifacts = filterArtifactsByType(
-    'system.ClassificationMetrics',
-    artifactTypes,
-    artifacts,
-  );
-
-  // There can be multiple system.ClassificationMetrics artifacts per execution.
-  // Get confidenceMetrics and confusionMatrix from artifact.
+  // There can be multiple system.ClassificationMetrics or system.Metrics artifacts per execution.
+  // Get scalar metrics, confidenceMetrics and confusionMatrix from artifact.
   // If there is no available metrics, show banner to notify users.
   // Otherwise, Visualize all available metrics per artifact.
-  const metricsAvailableArtifacts = getMetricsAvailableArtifacts(classificationMetricsArtifacts);
+  const verifiedClassificationMetricsArtifacts = getVerifiedClassificationMetricsArtifacts(
+    artifacts,
+    artifactTypes,
+  );
+  const verifiedMetricsArtifacts = getVerifiedMetricsArtifacts(artifacts, artifactTypes);
```

Comment:
In the future, could perhaps we can evolve the flow the following way:

1. Get collection of artifacts: {name, type, location}.
2. pass the collection to a function that
3. Iterates over the collection
4. `switch`es over the type
5. For known artifact types, constructs and adds visualization

Reply:
+1
I've previously gave the same comment and @zijianjoy agreed to do so later.
I'll treat as non blocker for this PR